### PR TITLE
Ensure pkg/install do not import pkg/plugin/framework

### LINKED ISSUE #8157: New dependencies relates to aws and azure sdk on Velero 1.14
**What steps did you take and what happened:**

Trying to use "install.VeleroOptions" to generate the velero option struct, then passing this as a parameter to the "install.Install" function. I was trying use the "install.Install" function to install Velero 1.14 on kubernetes cluster. 
https://github.com/vmware-tanzu/velero/blob/main/pkg/install/install.go

On the Velero 1.12, this method works for installing velero. But on Velero 1.14, it involves in lots of new dependency of "`github.com/aws/aws-sdk-go-v2`" and "`github.com/Azure/azure-sdk-for-go/sdk`". And the building does not work. Since we don't have this dependency on our repository yet.
https://github.com/vmware-tanzu/velero/blob/main/go.mod#L13

**What did you expect to happen:**
The Velero 1.14 server should install successfully using this package. 

**The following information will help us better understand what's going on**:

_If you are using velero v1.7.0+:_  
Please use `velero debug  --backup <backupname> --restore <restorename>` to generate the support bundle, and attach to this issue, more options please refer to `velero debug --help` 

_If you are using earlier versions:_  
Please provide the output of the following commands (Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)
- `kubectl logs deployment/velero -n velero`
- `velero backup describe <backupname>` or `kubectl get backup/<backupname> -n velero -o yaml`
- `velero backup logs <backupname>`
- `velero restore describe <restorename>` or `kubectl get restore/<restorename> -n velero -o yaml`
- `velero restore logs <restorename>`

**Anything else you would like to add:**

Here is the reason I found. The install package has dependency with these sdk.
`go mod why github.com/aws/aws-sdk-go-v2/credentials`
   `github.com/aws/aws-sdk-go-v2/credentials`
   `github.com/vmware-tanzu/velero/pkg/install`
   `github.com/vmware-tanzu/velero/pkg/repository`
   `github.com/vmware-tanzu/velero/pkg/repository/provider`
   `github.com/vmware-tanzu/velero/pkg/repository/config`
   `github.com/aws/aws-sdk-go-v2/credentials`

I knew in this release, Velero 1.14 including new features relates to aws and azure sdk. But on our dev side, I don't want to add too many new dependencies relates to these sdk for now. Any way to fix it or temporarily workaround? Thanks.

**Environment:**

- Velero version (use `velero version`): v1.14.0
- Velero features (use `velero client config get features`): 
- Kubernetes version (use `kubectl version`): Kubernetes v1.27.0
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`): SUSE Linux Enterprise Server 15 SP4

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

### PR DESCRIPTION
**Pull request by kaovilai**
Signed-off-by: Tiger Kaovilai <tkaovila@redhat.com>

Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #8157 (hopefully forever)
Rebased after #8145. This passes with #8145 and moving images.go away from /internal/velero which contain imports to pkg/plugin/framework which imports many cloud sdks. (created #8207)

This PR can be replaced or supplemented by #8208 

# Please indicate you've done the following:

- [ ] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Comment by kaovilai** [comment]
looks like 8145 ultimately failed to pass this test after some force pushes later. moving back to draft.

**Review by blackpiglet — DISMISSED** [review]

File: pkg/install/install_test.go [hunk 1]
```diff
@@ -147,3 +153,44 @@ func TestDaemonSetIsReady(t *testing.T) {
...
+	cmd.Dir = filepath.Dir(filename)
+	output, err := cmd.Output()
+	require.NoError(t, err)
+	// split dep by line, replace space with newline
+	deps := strings.ReplaceAll(string(output), " ", "\n")
+	require.NotEmpty(t, deps)
+	// ignore k8s.io
+	k8sio, err := regexp.Compile("^k8s.io")
+	require.NoError(t, err)
+	cloudProvider, err := regexp.Compile("aws|gcp|azure")
```
Comment (kaovilai) [inline pkg/install/install_test.go]:
```suggestion
	cloudProvider, err := regexp.Compile("aws|cloud.google.com|azure")
```

google regex fix

**Comment by ywk253100** [comment]
Closing this one as #8208 is merged

**Pull request closed without merging**